util/parsing.py

```diff
@@ -0,0 +1,79 @@
+"""
+This file contains some helper functions to parse python programs and visualize. They are not essential to the
+    reimplementation of ELM, but if one wants to generalize the traditional GP experiments to arbitrary functions,
+    hopefully this provides the code base for syntax parsing.
+For example, if we expand the TinyGP experiment to arbitrary functions, we can modify `GPTree` to non-binary trees,
+    and use the `tree.walk()` cursor demonstrated below to traverse (will also need to filter out some unwanted nodes).
+    The visualization code below will also be necessary for development and testing. But I am not sure if we want to
+    generalize the non-LM GP benchmarks. So I am leaving it here until the need arises in the future.
+"""
+
```

Comment:
A couple weeks ago I thought maybe we were gonna expand GP benchmark to something else and I wrote this up and tried some functions. But now I think the focus will probably really just be the language models. So feel free to delete these files in `util/`